gmapping/src/slam_gmapping.cpp

```diff
@@ -764,7 +771,7 @@ bool
 SlamGMapping::mapCallback(const std::shared_ptr<nav_msgs::srv::GetMap::Request> req,
                           std::shared_ptr<nav_msgs::srv::GetMap::Response> res)
 {
-  /* std::lock_guard<std::mutex> map_lock(map_mutex_); */
+  std::lock_guard<std::mutex> map_lock(map_mutex_);
```

Comment:
mop. Also a good catch. Probably should have uncommented that.

I haven't tested it, but this was formerly a `boost::scoped_mutex`. C++17 added a `std::scoped_mutex`, which we may consider changing to in the future, but I think the `std::lock_guard` will suffice.